src/main/java/spoon/reflect/visitor/ImportScannerImpl.java

```diff
@@ -218,7 +226,44 @@ private boolean addImport(CtTypeReference<?> ref) {
 		if (imports.containsKey(ref.getSimpleName())) {
 			return isImported(ref);
 		}
+		// don't import unnamed package elements
+		if (ref.getPackage() == null || ref.getPackage().isUnnamedPackage()) {
+			return false;
+		}
+		if (!ref.getPackage().getSimpleName().equals("java.lang")) {
+			if (classNamePresentInJavaLang(ref)) {
```

Comment:
first line is about that this class is not `java.lang`. Second line is about it's name is in java lang, i. e. com.mycompany.String.